git-branchless-invoke/src/lib.rs

```diff
@@ -117,12 +117,12 @@ fn install_tracing(effects: Effects) -> eyre::Result<impl Drop> {
 
 #[instrument]
 fn install_libgit2_tracing() {
-    fn git_trace(level: git2::TraceLevel, msg: &str) {
-        info!("[{:?}]: {}", level, msg);
+    fn git_trace(level: git2::TraceLevel, msg: &[u8]) {
+        info!("[{:?}]: {}", level, String::from_utf8_lossy(msg));
     }
 
-    if !git2::trace_set(git2::TraceLevel::Trace, git_trace) {
-        warn!("Failed to install libgit2 tracing");
+    if let Err(err) = git2::trace_set(git2::TraceLevel::Trace, git_trace) {
+        warn!("Failed to install libgit2 tracing: {err}");
```

Comment:
comment (non-blocking): In principle, we could also just forward the arguments like `warn!(?err, "Failed to install libgit2 tracing")` and get Structured Logging™ as well. I don't remember if there's a specific reason I wrote it this way, or if perhaps I wasn't acquainted with `tracing`. It's probably not important regardless.